packages/server/src/api/controllers/row/alias.ts

```diff
@@ -0,0 +1,125 @@
+import { QueryJson, SearchFilters, Table, Row } from "@budibase/types"
+import { getDatasourceAndQuery } from "../../../sdk/app/rows/utils"
+import { cloneDeep } from "lodash"
+
+export default class AliasTables {
+  character: string
+  aliases: Record<string, string>
+  tableAliases: Record<string, string>
+  tableNames: string[]
+
+  constructor(tableNames: string[]) {
+    this.tableNames = tableNames
+    this.character = "a"
+    this.aliases = {}
+    this.tableAliases = {}
+  }
+
+  getAlias(tableName: string) {
+    if (this.aliases[tableName]) {
+      return this.aliases[tableName]
+    }
+    const char = this.character
+    this.aliases[tableName] = char
+    this.tableAliases[char] = tableName
+    this.character = String.fromCharCode(char.charCodeAt(0) + 1)
+    return char
+  }
+
+  aliasField(field: string) {
+    const tableNames = this.tableNames
+    if (field.includes(".")) {
```

Comment:
I'm wondering if this is strictly valid. https://github.com/go-gorm/gorm/issues/1280 seems to suggest that `.` is allowed in column names. We may have to be smarter about how we split.